src/components/publishing/imageset_preview.tsx

```diff
@@ -1,106 +1,131 @@
 import React, { Component } from "react"
 import styled from "styled-components"
 
-import IconImageset from "./icons/icon_imageset"
-
 import Fonts from "./fonts"
+import IconImageSet from "./icons/icon_imageset"
 
-const Wrapper = styled.div`
-  max-width: 580px;
-  width: 100%;
-  height: 150px;
+const FullWrapper = styled.div`
+  position: absolute;
+  bottom: 20px;
+  left: 20px;
+  min-height: 50px;
+  width: auto;
+  max-width: calc(100% - 80px);
   display: flex;
+  justify-content: space-between;
+  align-items: center;
+  border: 1px solid white;
+  background: rgba(255, 255, 255, 0.8);
+  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.3);
+  padding: 20px;
+  `
+const MiniWrapper = styled.div`
+  height: 100px;
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  padding: 10px 20px 10px 10px;
+  border: 1px solid #e5e5e5;
 `
-const Text = styled.div`
-  ${Fonts.avantgarde("s11")}
-  line-height: 1.35em;
-  margin: 0;
-`
-const Remaining = styled.div`
-  min-width: 50px;
-  padding: 10px;
+const MiniInner = styled.div`
+  display: flex;
+  justify-content: space-between;
   flex: 1;
+  width: auto;
+  margin-left: 20px;
+`
+const TitleWrapper = styled.div`
   display: flex;
   flex-direction: column;
-  justify-content: center;
-  align-items: center;
-  border: 1px solid #e5e5e5;
-  text-align: center;
+  justify-content: space-between;
+  height: 100%;
+  &[data-layout='mini'] {
+    height: 50px;
+  }
+`
+const Title = styled.div`
+  ${Fonts.unica("s19", "medium")}
+  margin-bottom: 8px;
+  line-height: 1.1em;
+  `
+const SubTitle = styled.div`
+  display: flex;
+`
+const SubTitlePrompt = styled.div`
+  ${Fonts.unica("s14", "medium")}
+`
+const SubTitleCount = styled.div`
+  ${Fonts.unica("s14")}
+  margin-left: 20px;
 `
 const IconContainer = styled.div`
-  width: 32px;
-  margin-bottom: 10px;
+  height: 45px;
   position: relative;
+  margin-left: 20px;
+  text-align: right;
+  > svg {
+    height: 98%;
+  }
 `
 
-class ImagesetPreview extends Component<any, any> {
-  constructor(props) {
-    super(props)
-    this.state = {
-      visibleImages: this.getVisibleImages(this.props.images),
+class ImageSetPreview extends Component<any, any> {
```

Comment:
Typing the props would help the user of the component know how to use it more easily, especially with the `section.layout` prop, because it’s only 1 of 2 static strings and additionally they’d be told that the prop is required to specify. E.g.

```
interface Props extends {
  section: {
    layout: "mini" | "full"
    images: {
      url?: string
      image?: string
    }
  }
}

class ImageSetPreview extends Component<Props,null> {
  // ...
}
```

In the above example I only indicated with the second type parameter to `Component` that this class holds no state.